File: lessons/compiled_lang/cpp/src/j00.arguments.cpp
#include <cstdlib>
#include <iostream>

int main(int argc, char* argv[]) {
    int argCount = argc - 1;

    if (argCount != 2) {
        std::cerr << std::endl;
        std::cerr << "You need to enter two numbers:" << std::endl;
        std::cerr << std::endl;
        std::cerr << "   Usage: " << argv[0] << " [num1] [num2]" << std::endl;
        std::cerr << std::endl;
    } else {
        int sum = std::atoi(argv[1]) + std::atoi(argv[2]);
        std::cout << "The sum of " << argv[1] << " and " << argv[2] << " is: " << sum << "." << std::endl;
    }

    return 0;
}

File: lessons/compiled_lang/go/src/j00.arguments.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	scriptName := os.Args[0]
	args := os.Args[1:]

	if len(args) != 2 {
		fmt.Fprintln(os.Stderr)
		fmt.Fprintln(os.Stderr, "You need to enter two numbers:")
		fmt.Fprintln(os.Stderr)
		fmt.Fprintln(os.Stderr, "   Usage: "+scriptName+" [num1] [num2]")
		fmt.Fprintln(os.Stderr)
	} else {
		num1, _ := strconv.Atoi(args[0])
		num2, _ := strconv.Atoi(args[1])
		fmt.Printf("The sum of %s and %s is: %d.\n", args[0], args[1], num1+num2)
	}
}
